tinygrad/realize.py

```diff
@@ -25,7 +25,6 @@ def run_schedule(schedule:List[ScheduleItem], disable_logging=False):
     else:
       Device[si.out.device].exec_ast(si.ast, output=si.out, inputs=si.inputs, var_vals=si.var_vals, **si.out._device_extra_args())
     del si.out.op
-    for v in si.out.views: del v.op
```

Comment:
This is vital to making sure these ops are freed

Reply:
Ughh, that is disappointing, since this is a large burden to the runtime.

Let me think about another way to free them without referencing them.

Reply:
The biggest issue here is the creation of a slow `WeakSet()`. That can be avoided, see here https://github.com/tinygrad/tinygrad/pull/2364.

Reply:
Maybe, but the view would have to delete the op from the base if it realizes.